src/uu/dd/src/parseargs.rs

```diff
@@ -0,0 +1,761 @@
+            }
+            Self::NoMatchingMultiplier(arg) => {
+                write!(f, "Unrecognized byte multiplier -> {}", arg)
+            }
+            Self::ByteStringContainsNoValue(arg) => {
+                write!(f, "Unrecognized byte value -> {}", arg)
+            }
+            Self::MultiplierStringWouldOverflow(arg) => {
+                write!(
+                    f,
+                    "Multiplier string would overflow on current system -> {}",
+                    arg
+                )
+            }
+            Self::BlockUnblockWithoutCBS => {
+                write!(f, "conv=block or conv=unblock specified without cbs=N")
+            }
+            Self::StatusLevelNotRecognized(arg) => {
+                write!(f, "status=LEVEL not recognized -> {}", arg)
+            }
+            Self::Unimplemented(arg) => {
+                write!(f, "feature not implemented on this system -> {}", arg)
+            }
+        }
+    }
+}
+
+impl Error for ParseError {}
+
+/// Some flags specified as part of a conv=CONV[,CONV]... block
+/// relate to the input file, others to the output file.
+#[derive(Debug, PartialEq)]
+enum ConvFlag {
+    // Input
+    FmtAtoE,
+    FmtEtoA,
+    FmtAtoI,
+    Block,
+    Unblock,
+    UCase,
+    LCase,
+    Swab,
+    Sync,
+    NoError,
+    // Output
+    Sparse,
+    Excl,
+    NoCreat,
+    NoTrunc,
+    FDataSync,
+    FSync,
+}
+
+impl std::str::FromStr for ConvFlag {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            // Input
+            "ascii" => Ok(Self::FmtEtoA),
+            "ebcdic" => Ok(Self::FmtAtoE),
+            "ibm" => Ok(Self::FmtAtoI),
+            "lcase" => Ok(Self::LCase),
+            "ucase" => Ok(Self::UCase),
+            "block" => Ok(Self::Block),
+            "unblock" => Ok(Self::Unblock),
+            "swab" => Ok(Self::Swab),
+            "sync" => Ok(Self::Sync),
+            "noerror" => Ok(Self::NoError),
+            // Output
+            "sparse" => Ok(Self::Sparse),
+            "excl" => Ok(Self::Excl),
+            "nocreat" => Ok(Self::NoCreat),
+            "notrunc" => Ok(Self::NoTrunc),
+            "fdatasync" => Ok(Self::FDataSync),
+            "fsync" => Ok(Self::FSync),
+            _ => Err(ParseError::ConvFlagNoMatch(String::from(s))),
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Flag {
+    // Input only
+    FullBlock,
+    CountBytes,
+    SkipBytes,
+    // Either
+    #[allow(unused)]
+    Cio,
+    #[allow(unused)]
+    Direct,
+    #[allow(unused)]
+    Directory,
+    #[allow(unused)]
+    Dsync,
+    #[allow(unused)]
+    Sync,
+    #[allow(unused)]
+    NoCache,
+    #[allow(unused)]
+    NonBlock,
+    #[allow(unused)]
+    NoATime,
+    #[allow(unused)]
+    NoCtty,
+    #[allow(unused)]
+    NoFollow,
+    #[allow(unused)]
+    NoLinks,
+    #[allow(unused)]
+    Binary,
+    #[allow(unused)]
```

Comment:
Are these still unused?

Reply:
Removed for the most recent build. Looks like no errors were triggered.